waspc/data/Generator/templates/react-app/src/_Page.js

```diff
@@ -28,19 +35,36 @@ export class {= page.name =} extends Component {
 
   render() {
     return (
+      {=# wasp.isXRayModeEnabled =}
+      <>
+        <Switch
```

Comment:
Idea: move this into xRay/components/XRaySwitch or smth like that, and then just import it here. Make it already connected component, so you will not have to import state or actions here and code will be much simpler, here in the _Page, just import of one component and putting it in render().

Reply:
Great idea, did that.